PWGLF/TableProducer/lambdakzerobuilder.cxx

```diff
@@ -148,13 +160,225 @@ struct lambdakzeroBuilder {
     }
   }
 
-  void process(aod::Collision const& collision, aod::V0s const& V0s, MyTracks const& tracks
+  float getMagneticField(uint64_t timestamp)
+  {
+    // TODO done only once (and not per run). Will be replaced by CCDBConfigurable
+    static o2::parameters::GRPObject* grpo = nullptr;
+    if (grpo == nullptr) {
+      grpo = ccdb->getForTimeStamp<o2::parameters::GRPObject>("GLO/GRP/GRP", timestamp);
+      if (grpo == nullptr) {
+        LOGF(fatal, "GRP object not found for timestamp %llu", timestamp);
+        return 0;
+      }
+      LOGF(info, "Retrieved GRP for timestamp %llu with magnetic field of %d kG", timestamp, grpo->getNominalL3Field());
+    }
+    float output = grpo->getNominalL3Field();
+    return output;
+  }
+
+  void processRun2(aod::Collision const& collision, aod::V0s const& V0s, MyTracks const& tracks, aod::BCsWithTimestamps const&
 #ifdef MY_DEBUG
-               ,
-               aod::McParticles const& particlesMC
+                   ,
+                   aod::McParticles const& particlesMC
 #endif
   )
   {
+
+    float d_bz;
+    if (d_bz_input < -990) {
+      // Fetch magnetic field from ccdb for current collision
+      d_bz = getMagneticField(collision.bc_as<aod::BCsWithTimestamps>().timestamp());
```

Comment:
Won't this cause any overhead? I mean, maybe it could be useful to get the magnetic field only if the run number changed or something?

Reply:
Very reasonable suggestion, will implement run-switch-detector! Thanks!